### PR TITLE
*: all in camel case

### LINKED ISSUE #13216: Use camel case for struct naming

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

Issue Number: Close #13216
What's Changed:

Apparently the default clippy setting isn't doing much to detect the wrong naming. Update it and fix all issues.

```commit-message
```

### Check List 

Tests
- No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/error_code/src/engine.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ define_error_codes!(
     NOT_IN_RANGE => ("NotInRange", "", ""),
     PROTOBUF => ("Protobuf", "", ""),
     IO => ("IO", "", ""),
```
Comment (Connor1996) [inline components/error_code/src/engine.rs:9]: what about `IO`

File: components/raftstore/src/store/util.rs [hunk 2]
```diff
@@ -692,7 +692,7 @@ fn timespec_to_u64(ts: Timespec) -> u64 {
 ///
 /// # Panics
 ///
-/// If nsec is negative or GE than 1_000_000_000(nano seconds pre second).
+/// If nsec is negative or Ge than 1_000_000_000(nano seconds pre second).
```
Comment (Connor1996) [inline components/raftstore/src/store/util.rs:695]: greater than or equal to

**Review by Connor1996 — COMMENTED** [review]
How about enabling https://rust-lang.github.io/rust-clippy/master/index.html#use_self as well

File: components/error_code/src/engine.rs (hunk 1 above)
Comment (tabokie) [reply components/error_code/src/engine.rs:9]: No, it's a constant not a type.

**Comment by tabokie** [comment]
> How about enabling https://rust-lang.github.io/rust-clippy/master/index.html#use_self as well

I tried it, there're too many false positives, and a few regression IMO.
E.g.

```rust
     #[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq)]
     #[serde(rename_all = "kebab-case")]
64 | pub enum LogFormat {
   |          ^^^^^^^^^ help: use the applicable keyword: `Self`

          #[derive(Clone, Copy, Debug, PartialEq, Serialize)]
300 |     pub enum TestEnum {
    |              ^^^^^^^^ help: use the applicable keyword: `Self`

         fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
             match self {
31 |             ConfigValue::Duration(v) => write!(f, "{}ms", v),
   |             ^^^^^^^^^^^ help: use the applicable keyword: `Self`
```

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by sticnarf** [comment]
It may be still controversial to choose `TiDb` or `Tidb`. I find `Tikv` and `TiKv` are both used in the repo. I think we should unify them.

Personally, I prefer `Tikv` rather than `TiKV` because we usually use "tikv" as a word and if we camelize it, we will get `Tikv`.

**Comment by Connor1996** [comment]
> It may be still controversial to choose `TiDb` or `Tidb`. I find `Tikv` and `TiKv` are both used in the repo. I think we should unify them.
> 
> Personally, I prefer `Tikv` rather than `TiKV` because we usually use "tikv" as a word and if we camelize it, we will get `Tikv`.

vote for `Tikv`

**Comment by tabokie** [comment]
I'll change it. But it can't be checked with clippy though.

**Comment by Connor1996** [comment]
> I'll change it. But it can't be checked with clippy though.

use `disallowed_names`

**Comment by tabokie** [comment]
> use disallowed_names

I don't think so, it checks for exact match.

**Comment by sticnarf** [comment]
> Also, kvproto still uses TiDb (`ChangeDataRequestKvApi::TiDb`).

Go has a different naming convention. It's inevitable that we'll break the rule in some languages.

File: components/engine_traits/src/lib.rs [hunk 3]
```diff
@@ -191,7 +191,7 @@
 //!
 //! At the end of this phase the `engine` crate will be deleted.
 //!
-//! ## 3) "Pulling up" the generic abstractions through TiKv
+//! ## 3) "Pulling up" the generic abstractions through Tikv
```
Comment (sticnarf) [inline components/engine_traits/src/lib.rs:194]: It seems not a type name. `TiKV` should be used here, I think.

File: components/error_code/src/engine.rs [hunk 4]
```diff
@@ -7,7 +7,7 @@ define_error_codes!(
...
-    CF_NAME => ("CFName", "", ""),
+    CF_NAME => ("CfName", "", ""),
```
Comment (sticnarf) [inline components/error_code/src/engine.rs:10]: Will the change affect external components?

Comment (tabokie) [reply components/error_code/src/engine.rs:10]: I don't know. @5kbpers

**Review by sticnarf — APPROVED** [review]

**Pull request merged**